src/validation/utils.ts

```diff
@@ -1,80 +1,129 @@
 import { ValidationErrorMessage } from '../config'
 import { isIP } from 'net'
 
-export type Validator<V> = (value?: V) => ValidationErrorMessage
+export type Validator<V> = {
+  meta: {
+    min?: number
+    max?: number
+    validation?: string
```

Comment:
I think it's redundant to call this one validation, we're already within a Validator. I'd go back to sth like details or description